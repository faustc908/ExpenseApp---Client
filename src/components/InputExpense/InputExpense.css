.appDesc {
    border-top: 3px solid black;
    padding-top: 35px;
}

.enter{
    font-size: 15px;
    font-style: italic;
    padding-top: 10px;
}