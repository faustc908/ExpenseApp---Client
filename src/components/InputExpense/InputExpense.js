import React, { useState } from "react";
import config from "../config"

const InputExpense = ({ updateExpense }) => {
    const [description, setDescription] = useState("");

    // Post function to post expense notes

    const onSubmitForm = async (e) => {
        e.preventDefault();
        try {
            const body = { description };

            const header = {
                method: "POST",
                headers: { "Content-Type": "application/json" },
                body: JSON.stringify(body),
            };
            await fetch(`${config.API_ENDPOINT}/expense`, header).then(async function (
                response
            ) {
                await response.json();
                updateExpense();
            });
        } catch (err) {
            console.error(err.message);
        }
    };

    return (
        <div className="input">
            <h1 className="appDesc">User Notes</h1>
            <p>Please enter notes below</p>
            <form onSubmit={onSubmitForm}>
                <input
                    className="inputBox"
                    type="text"
                    value={description}
                    onChange={(e) => setDescription(e.target.value)}
                />
                <button className="btn" type="button" onClick={onSubmitForm}>
                    Add note
        </button>
            </form>
        </div>
    );
};

export default InputExpense;
