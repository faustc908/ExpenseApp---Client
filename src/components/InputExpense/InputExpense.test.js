import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import InputExpense from './InputExpense'
import ReactDOM from 'react-dom';

describe(`InputExpense component`, () => {

    it('renders a InputExpense by default', () => {
        const wrapper = shallow(<InputExpense />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})