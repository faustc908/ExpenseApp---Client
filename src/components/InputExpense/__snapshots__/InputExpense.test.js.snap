// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputExpense component renders a InputExpense by default 1`] = `
<div
  className="input"
>
  <h1
    className="appDesc"
  >
    User Notes
  </h1>
  <h4
    className="enter"
  >
    Please enter notes below:
  </h4>
  <form
    onSubmit={[Function]}
  >
    <input
      className="inputBox"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Add note
    </button>
  </form>
</div>
`;
